src/classes/TDTM_TriggerHandler.cls

```diff
@@ -216,6 +216,15 @@ public class TDTM_TriggerHandler {
         return null;
     }
 
+    private static Map<String, Type> typeMap = new Map<String, Type>();
+    private static Type getTypeForClassNamed(String str) {
+        if (typeMap.get(str) == null) {
```

Comment:
Just in case the class name that comes from the Trigger_Handler__c object is not always in the same case, maybe force it to lower case:
```suggestion
        str = str.toLowerCase();
        if (typeMap.get(str) == null) {
```